spec/main-spec.rb

```diff
@@ -0,0 +1,104 @@
+require "./lib/logic.rb"
+
+
+RSpec.describe Board do
+      before do
+        @board = Board.new
+      end
+      describe "#win_check" do
+      it "checks if player has won by row" do
+        @board.current_state = [["X","X","X"],["2,1","2,2","2,3"],["3,1","3,2","3,3"]]
+        expect(@board.win_check("X")).to eql(true)
+
+        @board.current_state = [["1,1","1,2","1,3"],["X","X","X"],["3,1","3,2","3,3"]]
+        expect(@board.win_check("X")).to eql(true)
+
+        @board.current_state = [["1,1","1,2","1,3"],["2,1","2,2","2,3"],["X","X","X"]]
+        expect(@board.win_check("X")).to eql(true)
+      end
+
+      it "checks if player has won by column" do
+        @board.current_state = [["X","1,2","1,3"],["X","2,2","2,3"],["X","3,2","3,3"]]
+        expect(@board.win_check("X")).to eql(true)
+
+        @board.current_state = [["1,1","X","1,3"],["2,1","X","2,3"],["3,1","X","3,3"]]
+        expect(@board.win_check("X")).to eql(true)
+
+        @board.current_state = [["1,1","1,2","X"],["2,1","2,2","X"],["3,1","3,2","X"]]
+        expect(@board.win_check("X")).to eql(true)
+      end
+
+      it "checks if player has won by diagonals" do
+        @board.current_state = [["X","1,2","1,3"],["2,1","X","2,3"],["3,1","3,2","X"]]
+        expect(@board.win_check("X")).to eql(true)
+
+        @board.current_state = [["1,1","1,2","X"],["2,1","X","2,3"],["X","3,2","3.3"]]
+        expect(@board.win_check("X")).to eql(true)
+      end
+    end
```

Comment:
About testing `Board` class
- Test the initialize method
- Add negative scenario

Reply:
What about the **Draw** state ?